### DISCUSSION TITLE
Wire:navigate not working when applying hashtags

### DISCUSSION THREAD

**Discussion by marknuyens** [livewire/livewire#8073] [7. Bugs (archived)] [answered]
### Livewire version

v3.2.6

### Laravel version

v10.0.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. Create a link that contains a hashtag
2. Apply `wire:navigate` to that link.
3. Click on the link and watch your address
4. Notice how the hash has not been applied

### Repository or Wirebox URL

/

### Additional information

It stands to reason that this should be possible, given how the page itself can be fetched and after it has been fetched, the hashtag should be applied as one would expect. I could not find any mention of this in the [docs](https://livewire.laravel.com/docs/navigate#customizing-the-progress-bar) and if this is not supported for some reason, I would expect it to be included.

At the moment, I am simply using a full page refresh instead, but of course feels quite dirty compared to Livewire's clean solution through `wire:navigate`.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@marknuyens Can you please share an example link tag code with what you mean? Thanks!

**Reply by danie-ramdhani** [reply]
@joshhanley maybe it look like this https://wirebox.app/b/x2zpz

- user click link with hashtag
- when user navigated to the link, the hashtag is disappear

**Reply by marknuyens** [reply]
Yes, this should represent the issue. Given how reproducing the issue was relatively easy, I figured an entire replication was not necessary.

I hope this will suffice, and look forward to your response and possible solution.

**Reply by joshhanley** [reply]
@danie-ramdhani thanks, that's great!

@marknuyens I wasn't sure if you were talking about an internal page anchor link (for example `<a href="#test" >`) or a different page anchor link like Danie demonstrated. Sorry about my brevity - I was replying from my phone at the time.

When reporting it's always best to share code, doesn't need a full reproduction if it's a simple one line but sharing the code for the `<a>` tag so there is no ambiguity is helpful 🙂 

Of course any reproduction in something like wirebox is even better for us, as we can copy it directly into a test so we can investigate it and get it fixed. The ultimate for us though is if people can submit failing test PRs as it allows us to just focus on fixing the problem and not having to worry about whether we can replicate the problem properly or not.

Regarding this issue, yeah I don't think `wire:navigate` currently supports anchor tags like that on different pages (probably just an oversight). It's not going to be trivial to add, so it probably won't happen quickly. If you have anchor tags on the same page though I wouldn't be adding `wire:navigate` to the `<a>` tag.

**Reply by marknuyens** [reply]
You are entirely correct by mentioning I should have been more specific. I realize that I didn't really indicate the exact circumstances in which my issue occurred. I do, however, feel this issue should deserve some attention. Referring to a different page is trivial and adding a hashtag would make sense, especially if you're redirecting users to a page that uses tabs and those tabs are connected by hashtags, as applies in my case.

Anyway, it does qualify as a bug, so should I `create issue from discussion` or will you take it from here?

**Reply by joshhanley** [reply]
@marknuyens we are now using discussions to track bugs and ideas so this is the bug report.
See this announcement for more details https://github.com/livewire/livewire/discussions/3147

If you have the time, we'd really appreciate a failing test PR that demonstrates the issue 🙂 A failing test PR is also the best way to get Caleb to look at it.

**Reply by danie-ramdhani** [reply, marked as answer]
Made failing test PR https://github.com/livewire/livewire/pull/8143

**Reply by marknuyens** [reply]
Thank you. 🤓

**Reply by danie-ramdhani** [reply]
PR is merged with fix bug, should be fixed in next release.

**Reply by danie-ramdhani** [reply]
do you like podcast @marknuyens ? this one is interesting 😁👍 (https://notesonwork.transistor.fm/episodes/urls-and-dramatic-movies).

**Reply by marknuyens** [reply]
Yes, I do! 🤓 In fact, I’m building my own podcasting platform as we speak — using Livewire, of course.

Anyway, just listened to the episode, interesting to hear how the bug was apparently not as trivial as I thought it was. 😅

**Reply by joshhanley** [reply]
@marknuyens yeah the internals of Livewire can lead to some interesting and unexpected rabbit holes at times 😆
